CITATION.cff

```diff
@@ -0,0 +1,75 @@
+################################# MIT LICENSE ##################################
+#
+# MIT License
+#
+# Copyright (c) 2022 Pablo Hamacher
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+################################################################################
+
+################################################################################
+##
+##  AUTHOR      Kevin Matthes
+##  BRIEF       Citation settings for the Git repository.
+##  COPYRIGHT   (C) 2022 Kevin Matthes.
+##              This file is licensed MIT.
+##  DATE        2022
+##  FILE        CITATION.cff
+##  NOTE        See `LICENSE' for full license.
+##              See `README.md' for project details.
+##
+################################################################################
+
+# Parser settings.
+cff-version: 1.2.0
```

Comment:
I don't think this is needed for this small project. Like I already said in the issue, you can cite it anyway you want, you could also drop any citation, since I really do not care.

I think adding another file will just cluster this repository and make it less easy to understand what is actually important.

Reply:
There are multiple approaches to give your users an overview about the sources of knowledge and / or inspiration you used to create your project.  CFF is analogue to a list of references in a paper.

And:  it can be automised.  I created myself a tool, `cffreference`, to extract the citation information of a given `CITATION.cff` to append if to another one in its `references` section.  That way, I just need to run a few Haskell commands in `GHCi` over my references section to hand it over to `cffconvert` which creates me a BibTeX database of software citations.  (There is a little more string processing with Haskell in between but basically, this is the way to go to get the BibTeX entries of 100 and more references in under 5 minutes.)

And I always cite anything.  Even dictionaries.  And formula collections.  By that, you will never forget any reference.

Finally, the day will come you will need to refer an algorithmic idea or an implementation technique in one of your papers.  `cffconvert` will give you your BibTeX entry in less than half a minute.

Reply:
If you consider the GitHub Action too much, then okay, I can remove it.  But the `CITATION.cff` should be a part of this project after all.

If you disagree with the metadata I suggested, you are free to change it.

Any mental property should be cited when used.  With this file, you tell others (and GitHub) how you would like your work to be referenced.  You are the author so you have the right to define the way how people should refer to your project.

Reply:
The thing is, I do not care at all, how or even if, others are referencing this or not. I, as the author, tell the people, that they should refer to the project in any way they like.

And if anyone disagrees with this, they can open their own fork and dictate citation in whatever way they like. 😄

Reply:
Hmm, I still think, that `CITATION.cff` should go as well. It is just not as important for a project, like the `README.md` or the `LICENSE`, especially with this being a small hobby project. 😅

Reply:
GitHub once had a list of recommended community standards it showed for each project in the Insights tab.  But due to an update, it moved somewhere else.

This list contained items such as `CONTRIBUTING.md` or a Code of Conduct.  I am unsure if it nowadays also contains the CFF as recommendation.

Do you know where this list moved to?  It might be a more reliable guideline to answer this question.

Reply:
Never mind, I just found it.  Looks like you need to be in an independent repository and not in a fork.  It is still named Community Standards and can be found in Insights.

But unfortunately, it does not contain a CFF as recommendation.

Reply:
Sadly I have no idea.

But again, I think the size of this project would be to small for a `CONTRIBUTING.md` or a Code of Conduct. 😅

Reply:
When considering the recommended things in contrast to the CFF, which would you consider more important -- from all of those points?

The recommended things are (checked means that this repository already has it):

* [x] README.md
* [x] description
* [x] LICENSE
* [ ] CODE_OF_CONDUCT.md
* [ ] CONTRIBUTING.md
* [ ] Issue templates
* [ ] Pull Request templates

Maybe CFF will be added in a later version.

When choosing from this list and the already suggested CFF, which would be the most important one to add next if you would need to add one?

Reply:
Even a small hobby project can become important enough to need to cite it.  Especially if you activate the career option of GitHub, this will become very important since you are applying with your repositories as examples of which masterpieces of software you already crafted.

Reply:
Tbh, I don't think any of this need to be added in the near future. There are neither enough Issues nor Pull Request that would make templates necessary.

The same goes for Contributing, you would be the first contributor and, I suspect, the only one for a long time.

And while a code of conduct would be nice, there is a simple cargo command (`cargo fmt`) which would suffice for that. Problem is, I did not use it explicitely during this project, so this code would probably go against that code of conduct already.. 😅

Reply:
Okay, so the CFF should be removed for now and added when required in the future?

Reply:
Yeah, I think that would be the right course of action.

Tbh I would probably concentrate on some of my bigger projects on github (and some currently private ones) for job hunting. But this is right now also no priority of mine. 😅

Reply:
Fine, the CFF is gone.  By the way, I considered the GitHub default "Delete CITATION.cff" descriptive enough to leave as commit message.